numpy/core/einsumfunc.py

```diff
@@ -1358,11 +1358,22 @@ def einsum(*operands, out=None, optimize=False, **kwargs):
         raise TypeError("Did not understand the following kwargs: %s"
                         % unknown_kwargs)
 
-
     # Build the contraction list and operand
     operands, contraction_list = einsum_path(*operands, optimize=optimize,
                                              einsum_call=True)
 
+    # Handle order kwarg for output array, c_einsum allows mixed case
+    original_order = kwargs.pop('order', 'K')
+    output_order = original_order.upper()
+    if output_order not in ['C', 'F', 'A', 'K']:
+        raise ValueError(f"order must be one of 'C', 'F', 'A', or 'K' (got '{original_order}')")
```

Comment:
Hmm, did not make up my mind yet, but should we just delete this here for simplicity? On the off-chance there is an error, I doubt it matters much if the error is given after the calculation is finished?

Reply:
Happy too. As mentioned I don't have strong opinions, I just try to reproduce `c_einsum` error messages verbatim where I can.

Reply:
The error message will be identical in either case (all those functions use the same parser internally).  The only advantage is to catch the error early, and I am not sure I feel that is worth the clutter?

Reply:
Will do, though there is an interesting optimization that causes the current exception test to fail:
```
>>> np.asanyarray(np.arange(6), order='bad order')
array([0, 1, 2, 3, 4, 5])
```

Reply:
I agree its a bug... The optimization is only worth it, because our kwarg-parsing is slow, I have a speed up for that, but its a large PR and it never got much attention.
In any-case, its a bug in `np.array` and friends, not here...

Reply:
Yea, no harm really. I reshaped the array so it will properly catch. I think this is likely good to go now.

Reply:
Yes, thanks for the beautiful tests as well and the quick followups! I am just waiting for the tests at this time.

Reply:
Actually, I have one cosmetic thing, changing `flags.["F_CONTIGUOUS"]` to simply use `flags.f_contiguous`.  But I can just fix that up myself as well.